projects/module-1/which-beyonce-group.md

```diff
@@ -0,0 +1,220 @@
+
+## Progression
+
+### Iteration 0 - Initial Layout & UI
+
+This should be completed by the end of the first day.
+
+- Build out views for both the game page and win page (this should all be in a single HTML file)
+- Build out the HTML for the cards to figure out how to scatter them (see the comp)
+- BUT! Know that this hardcoded HTML will not be used; the cards will be programmatically created by the JavaScript in a later iteration
+- Get at least a rough draft of your styling finished.
+
+### Iteration 1 - Setting Up the Data Model
+
+So what IS the data model?
+
+In most complex applications, user inputs and moving pieces of information are handled by JavaScript, and stored in complex data types (arrays, objects). Once the data has been updated, _then_ the program uses those arrays and objects to display the new information on the DOM.
+
+This is done for a few reasons:
+
+- We never want to show information to the user that is not accurate
+  - For example, if you Liked a tweet on Twitter, and you saw the red heart appear, you would expect that your Like had been saved, and if you refreshed the page, it would still show up
+  - But what if Twitter built their logic to update the DOM first, and then try to save that info?
+  - It would be possible to see a change, but for that change to not be stored anywhere.
+  - We want to avoid that confusing situation!
+- We are always going to store information in JS objects and arrays FIRST
+- Then, we'll display that information on the DOM by iterating over our stored data
+
+Sometimes you will hear this described as separating "business logic" from "display logic". It's also known as "separation of concerns" - that is, making sure that we're always updating our source of truth and then using it to display information on the DOM.
+
+To do this, we're going to create two JavaScript classes: `Card` and `Deck`.
+
+<section class="call-to-action">
+### Data:
+
+- Each card should get its data from an object instance of a Card class
+- Each card should have a `matchInfo` property which will hold the same value as the card that matches it. You can determine how to format that matching value. Each card should also have a `matched` property, a boolean, that will default to false and toggle to true once a match has been made.
+- Each card should have a method `match` on it
+- All card object instances should be held in the *Deck* class
+- The Deck should have `cards`, `matchedCards`, and `selectedCards` as properties. The Deck should have the following methods: `shuffle`, `checkSelectedCards`, and `moveToMatched`. If you find a need for additional methods, write/use them!
+</section>
+
+### Iteration 2 - Rendering the Data Model
+
+- Iterate over the `Deck`'s array of `Card` object instances to display the cards on the DOM (for now, don't worry about scrambling up the cards)
```

Comment:
I'm tempted to make this less specific so they have more heavy lifting. Like - "Display cards on the DOM that represent the data in your card object instances."

???

Reply:
I kind of wanted to explicitly tell them what to do in this instance, since there will be plenty of data model work that they'll have to figure out in following iterations ....